### PR TITLE
CVE-2014-0231

### PR DESCRIPTION
**Pull request by etiano417**
- Filled yml fields for CVE-2014-0231
 - replaced incorrect fix commits with accurate ones for CVE-2014-0231

### CODE REVIEW

File: cves/CVE-2014-0231.yml [hunk 1]
```diff
@@ -61,19 +74,21 @@ bounty:
...
+   - commit: 41488e891d12eddcf21bb435d90ae71eda8d218a
+     note: Introduce a read timeout to prevent scripts from hanging too long
+   - commit: c0be5abcd369a29528c9e9f3cc5b74753a3ec4a5
+     note: Introduced timeouts for scripts that don't read from stdin
 vccs:
-  - commit:
-    note:
+  - commit: 9ca05a3a514cf399d604a33b37dbc18e4fcd886a
+    note: |
+      Because the vulnerability is caused by a lack of measure against halting,
```
Comment (well-mannered-rascal) [inline cves/CVE-2014-0231.yml]: Run on sentence, breaking it up into two parts might help readability.

File: cves/CVE-2014-0231.yml [hunk 2]
```diff
@@ -177,38 +207,43 @@ lessons:
...
-    applies:
+    applies: false
     note:
   distrust_input:
-    applies:
-    note:
+    applies: true 
+    note: |
+      centered around handling abnormal input (input which causes the script 
+      to halt)
```
Comment (well-mannered-rascal) [inline cves/CVE-2014-0231.yml]: I think that having a full sentence and a little more detail here would be important for the CVE.

File: cves/CVE-2014-0231.yml [hunk 3]
```diff
@@ -177,38 +207,43 @@ lessons:
...
-    applies:
+    applies: false
     note:
   complex_inputs:
-    applies:
-    note:
+    applies: true
+    note: |
+      premised on the idea that CGI script writers may not anticipate the 
+      impact of certain inputs
```
Comment (well-mannered-rascal) [inline cves/CVE-2014-0231.yml]: Lacking a subject of the sentence here, again perhaps a little more detail and a full sentence would add more value to the CVE

File: cves/CVE-2014-0231.yml [hunk 4]
```diff
@@ -222,4 +257,17 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+  
+    This is a particularly interesting case of finding a way to make the system 
+    fail gracefully. The issue - the process responsivle for running CGI scripts
```
Comment (well-mannered-rascal) [inline cves/CVE-2014-0231.yml]: Small spelling error here on "responsive"

**Comment by well-mannered-rascal** [comment]
Overall, well done in my opinion. I give this CVE 2 upvotes. (I would give more but I was not careful and gave two many away already).

**Comment by etiano417** [comment]
Fixed issues noted by @heartemoji (see the previous commit)

File: cves/CVE-2014-0231.yml [hunk 5]
```diff
@@ -48,7 +48,20 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+
+    A CGI (Common Gateway Interface) is a protocol used by web servers that
+    runs command-line interface scripts in response to client requests. Apache 
+    HTTP Server has a module "mod_cgid" which is responsible for runnig CGI
```
Comment (ghost) [inline cves/CVE-2014-0231.yml]: Typo: runnig -> running

File: cves/CVE-2014-0231.yml [hunk 6]
```diff
@@ -61,19 +74,20 @@ bounty:
...
+     note: Introduce a read timeout to prevent scripts from hanging too long
+   - commit: c0be5abcd369a29528c9e9f3cc5b74753a3ec4a5
+     note: Introduced timeouts for scripts that don't read from stdin
 vccs:
-  - commit:
-    note:
+  - commit: 9ca05a3a514cf399d604a33b37dbc18e4fcd886a
+    note: |
+      The vulnerability was introduced with the mod_cgid component (which lacked
+      proper measures against halting from its introduction).
```
Comment (ghost) [inline cves/CVE-2014-0231.yml]: I would use the same wording from the description. You could say that the component lacked command timeout from its introduction instead.

File: cves/CVE-2014-0231.yml [hunk 7]
```diff
@@ -117,9 +134,15 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer:
-  date:
-  automated:
+  answer: |
+    Rainer Jung, a member of the Apache Software Foundation, discovered this
+    vulnerability on June 16th 2014 and reported it to the Apache Security Team.
```
Comment (ghost) [inline cves/CVE-2014-0231.yml:145]: Did he say how he discovered it?

File: cves/CVE-2014-0231.yml [hunk 8]
```diff
@@ -177,38 +206,46 @@ lessons:
...
   complex_inputs:
-    applies:
-    note:
+    applies: true
+    note: |
+      This vulnerability depends on the fact that CGI script writers may not 
+      anticipate the impact of certain inputs. The complexity of the 
+      relationship between script inputs and execution time means that it may
+      not be apparent to a CGI script writer whether their script will run
+      without halting on all conditions.
```
Comment (ghost) [inline cves/CVE-2014-0231.yml]: In this case, aren't CGI script writers users of the system? From the perspective of the contributors to httpd, isn't the input just a script that runs for some unknown amount of time? This should be written from the perspective of httpd's contributors.

File: cves/CVE-2014-0231.yml [hunk 9]
```diff
@@ -177,38 +206,46 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies:
-    note:
+    applies: true
+    note: |
+        timeouts are a layer of robustness underneath normal CGI input validation
```
Comment (ghost) [inline cves/CVE-2014-0231.yml]: Missing capitalization and a period.

File: cves/CVE-2014-0231.yml [hunk 10]
```diff
@@ -4,7 +4,7 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE:
+CWE: 754
```
Comment (ghost) [inline cves/CVE-2014-0231.yml]: CWE-754: Improper Check for Unusual or Exceptional Conditions

I would think that a non terminating script is something to be expected for a scriptable system.

I think CWE-400: Uncontrolled Resource Consumption might be better.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
1 upvote

File: cves/CVE-2014-0231.yml (hunk 6 above)
Comment (etiano417) [reply cves/CVE-2014-0231.yml]: Thanks for the recommendation! I'll definitely use this change of wording.

File: cves/CVE-2014-0231.yml (hunk 10 above)
Comment (etiano417) [reply cves/CVE-2014-0231.yml]: good catch! I think that is more appropriate, thank you.

**Pull request merged**